builtin/logical/pki/path_intermediate.go

```diff
@@ -105,6 +97,10 @@ func (b *backend) pathGenerateIntermediate(ctx context.Context, req *logical.Req
 		}
 	}
 
+	if err = parsedBundle.CSR.CheckSignature(); err != nil {
```

Comment:
I'd prefer to hoist this into `certutil` closer to where we generate the CSR (`createCSR`) if we can? That way callers of the SDK aren't affected by the same bug too.